libutils/unix_dir.c

```diff
@@ -194,8 +194,10 @@ static size_t GetDirentBufferSize(size_t name_len)
 
 struct dirent *AllocateDirentForFilename(const char *filename)
 {
-    struct dirent *entry = xcalloc(1, GetDirentBufferSize(strlen(filename)));
+    int length = strlen(filename);
+    struct dirent *entry = xcalloc(1, GetDirentBufferSize(length));
 
-    strcpy(entry->d_name, filename);
+    // d_name is fixed length, but we have allocated extra space using xcalloc:
+    strncpy((char *)entry->d_name, filename, length + 1);
```

Comment:
`strncpy` is generally considered bad practice for string copying, slow and easy to get it wrong. Here all you need is `memcpy` or `strcpy`, or if you want to be extra pedantic go with `strlcpy`.

Reply:
The "fix" here is basically the cast, so add to your comment that this cast is used to silence compiler warnings.

Reply:
Thanks, used memcpy instead.